app/models/publication.rb

```diff
@@ -149,22 +149,24 @@ def update_from_pubmed
     pm_source_record = PubmedSourceRecord.find_by_pmid(pmid)
     return false unless pm_source_record
 
-    pm_source_record.pubmed_update
-    rebuild_pub_hash
+    pm_source_record.pubmed_update # update the source data via the Pubmed API
+    self.pub_hash = pm_source_record.source_as_hash # update our local pubhash
+    pubhash_needs_update!
+    save
   end
 
   # @return [Boolean] true if .save is successful
   def rebuild_pub_hash
     raise 'rebuilding WOS records is unsupported, unimplemented' if wos_pub?
 
-    if sciencewire_id
+    if sciencewire_pub?
       sw_source_record = SciencewireSourceRecord.find_by_sciencewire_id(sciencewire_id)
       build_from_sciencewire_hash(sw_source_record.source_as_hash)
-    elsif pmid
+    elsif pubmed_pub?
       pubmed_source_record = PubmedSourceRecord.find_by_pmid(pmid)
```

Comment:
using these helper methods check the publication provenance instead of just looking for the presence of an identifier, which is safer